### DISCUSSION TITLE
Lights in one area always seem to be on when they shouldn't be

### DISCUSSION THREAD

**Discussion by ReplicaObscura** [jseidl/magic-areas#249] [Q&A] [answered]
I have all of my areas configured basically the same--the lights come on when occupied or extended, and the sleep lights come on during sleep.

However, my living room lights seem to be on every morning when I come downstairs (even though they are turned off at night during sleep properly), and tend to stay on all day unless I manually turn them off.

I thought maybe it was because I was also using the "playing" state to determine occupancy and one of my media players was misbehaving, but I removed that option so it only uses "on" and "open" states and the lights are still on all the time.

During troubleshooting, I also completely removed the dark state and accented state entities so that the logic is simpler, but the issue still persists.

Are there common causes for that behavior to occur? I don't have any other automations or external controls turning the lights on so I think it has to be MA, but I'm having trouble determining the reason.

**Comment by jseidl** [comment]
Hmmm this is very weird. What could be happening is that one of the presence sensors is either on all the time or flapping, keeping the area occupied. What the area occupancy binary sensor history looks like? It's been on all the time?

If you see the attributes of that `binary_sensor.area_$area_name`, the attribute `active_sensor` will tell if there's a presence sensor active.

Also, when this happens, are the lights on with the area occupied or clear?

**Reply by ReplicaObscura** [reply]
I thought I figured this out... but now two rooms are still detecting presence all the time and I don't know why.

For instance, my living room. The Area entity says "Detected" 2 hours ago, which was when sleep mode turned off, and it has been solid On since then. However, the "Active sensors" attribute is empty so I'm confused about why it's still showing presence.

I do have area light control turned on, and area presence hold turned off.

I seem to have the same issue in my studio, where it holds presence even when there are no active presence sensors and presence hold is off. I feel like I must be missing something.

Update: In my living room, clicking Configure and re-saving all the existing Magic Area settings (without making any changes), and then waiting for the presence timeout, the area finally turned off presence. So even though nothing was indicating presence, the area held presence from when sleep mode turned off until I re-saved all settings.

I have a feeling the studio issue is similar, and that if I re-save all my settings it'll reset the area presence after the timeout.

Is it possible there is some bug that makes it so turning off sleep mode can permanently set the room as occupied regardless of other sensors? Maybe it's relevant that both rooms were "occupied" at the time I switched sleep mode to On?

**Reply by ReplicaObscura** [reply]
This issue is continuing to occur... two of my rooms are sometimes keeping the lights on indefinitely long after the last presence detection event timeout. Seemingly for just these two rooms, and it happens more often in the living room than the studio.

Both rooms have more presence sensors than my other rooms, so maybe it is a matter of complexity?

Sometimes the Area entity will show a presence sensor as active which should no longer be active, but other times it shows no presence sensors as active yet still shows the area as occupied.

The only way to fix it when this happens seems to be to re-configure that Magic Areas entity... keeping all settings the same, then when submitting the final screen, the area entity properly stops showing it's occupied.

**Reply by jseidl** [reply, marked as answer]
Hey Ben,

Sorry for the delay on getting back to you. This week has been incredibly crazy!

> Sometimes the Area entity will show a presence sensor as active which should no longer be active

Hmm. So the area tracks entities over entity state change callbacks from Home Assistant. As a safe-measure, in case one of those callbacks are missed somehow, all entities are checked on `update_interval`. Might be that one of those state change events from HA didn't reached out MA. When you say "should no longer", does it mean the sensor's entity in Home Assistant was `off` and it was still showing in the area as an `active_sensor`?

> (...) other times it shows no presence sensors as active yet still shows the area as occupied.

The area will always wait `timeout_seconds` since the last precence sensor returned into the `off` state before clearing. 

> Both rooms have more presence sensors than my other rooms, so maybe it is a matter of complexity?

It's hard to say. I'd add all those entities that are being used as `presence_sensors` into Home Assistant's History along with the area binary sensor. That would help see if there's an entity giving false-positives and keeping the area occupied. I really should add some new attributes like `last_triggered_entity_id` and `last_triggered_time` to help debug those issues.

> The only way to fix it when this happens seems to be to re-configure that Magic Areas entity... keeping all settings the same, then when submitting the final screen, the area entity properly stops showing it's occupied.

Interesting. Re-configuring the area does reload it fully. If it's not coming back already (or changing shortly after coming up) as occupied, then either something is flapping or very weird. If for any error the code crashed, the entity would go `Unavailable`, and not be stuck in a state, so I'm very curious about what is going on. 

We can do a log dive if needed but it's pretty intense (the debug log is preeetty verbose) but I think we can solve by looking at the entities through the history and checking the states of all of them side by side. Can you do that and let me know what you see?

Cheers!

**Reply by ReplicaObscura** [reply]
No prob, thanks for the reply!

> When you say "should no longer", does it mean the sensor's entity in Home Assistant was off and it was still showing in the area as an active_sensor?

I need to double check this just in case the motion sensor was somehow misbehaving and staying on indefinitely. If I see that behavior again I'll double check and let you know.

> The area will always wait timeout_seconds since the last precence sensor returned into the off state before clearing.

My extended wait timeout is 300 seconds, and the Area entity stayed On for hours with no sensors listed in Active. The only way I could figure out how to reset it was to reconfigure the area. This seems to happen to both of my more complex areas (studio and living room), but most often in the living room.

I'm still trying to figure out additional details on my end, too. After the last time I re-saved both entities, they've been working properly for a couple of days now, so it's possible it won't occur again. If it does, I'll inspect the state of all the entities in the history and follow up again.

Thank you!

**Comment by ReplicaObscura** [comment]
Ah, that does seem related. The binary sensor every once in a while resets itself to "on" but never turns off, it's just solid on. It's using the same type of Aqara motion sensor that most of my other rooms have, although I have a LOT of media players and other sensors in the room that are also being taken into account with that binary sensor so one of them may be misbehaving

**Comment by ReplicaObscura** [comment]
Aha! I created a binary sensor, perhaps counter-intuitively, to indicate when I am NOT watching TV, for light automation use, but didn't think about it until now that it's using that binary sensor as a presense sensor, so anytime I am not watching TV, there is presence in the room.

I'm going to exclude that sensor and I am betting that'll fix things. Thanks for pointing me in the right direction!
